src/main/kotlin/org/jetbrains/tinygoplugin/services/TinyGoInfoExtractor.kt

```diff
@@ -0,0 +1,65 @@
+package org.jetbrains.tinygoplugin.services
+
+import com.goide.util.GoExecutor
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import org.jetbrains.tinygoplugin.GarbageCollector
+import org.jetbrains.tinygoplugin.Scheduler
+import org.jetbrains.tinygoplugin.TinyGoConfiguration
+import java.nio.file.Paths
+
+class TinyGoInfoExtractor(private val project: Project) {
+    companion object {
+        private val logger: Logger = Logger.getInstance(TinyGoInfoExtractor::class.java)
+    }
+
+    fun assembleTinyGoShellCommand(): GoExecutor {
+        val executor = GoExecutor.`in`(project, null)
+        val settings = TinyGoConfiguration.getInstance(project)
+        val parameters = tinyGoArguments(settings)
+        executor.withParameterString("info $parameters")
```

Comment:
You can pass parameters as a list of strings. In this case, you won't need to join them before ahead.